AutoGrade/models.py

```diff
@@ -123,17 +123,20 @@ def moss_report(self):
     def moss_submit(self):
         moss_folder = 'uploads/moss_submission/assignment_{0}/'.format(self.id)
 
-        submissions = self.get_student_latest_submissions()
-
-        if not submissions:
-            logging.debug("MOSS: No submissions available for generating Moss report")
-            return False
+        submissions = self.get_student_latest_submissions() # include all students of the assignment's course annd submission can be empty
 
+        submission_count = 0
         for submission, student in submissions:
-            modifiable_file = submission.get_modifiable_file()
-            path = moss_folder + basename(modifiable_file).replace(".", "-" + submission.student.get_roll_number() + ".")
-            touch(path)
-            copyfile(modifiable_file, path)
+            if submission == None:
```

Comment:
`== None` ? Or am I just not awake yet?

Reply:
@recluze `get_student_latest_submissions` returns submissions like [(submission, student1),(submission, student2)...]

There can be entries, which will have student information but no submission eg. [...(None, student4),(submission, student5,)...]. So just added condition for that on line 130.

Will create issue (enhancement) to remove: student from that tuple :) it doesn't looks good now.

Reply:
Hnn ... please go ahead and commit. That enhancement will break a piece of code I'm working on but I can fix that later...